Source/santad/ProcessTree/SNTEndpointSecurityAdapter.mm

```diff
@@ -0,0 +1,70 @@
+/// Copyright 2023 Google LLC
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+#include "Source/santad/ProcessTree/SNTEndpointSecurityAdapter.h"
+
+#include <EndpointSecurity/EndpointSecurity.h>
+#include <Foundation/Foundation.h>
+#include <bsm/libbsm.h>
+
+#include "Source/santad/EventProviders/EndpointSecurity/EndpointSecurityAPI.h"
+#include "Source/santad/ProcessTree/process_tree.h"
+#include "Source/santad/ProcessTree/process_tree_macos.h"
+#include "absl/status/statusor.h"
+
+using santa::santad::event_providers::endpoint_security::EndpointSecurityAPI;
+
+namespace santa::santad::process_tree {
+
+void InformFromESEvent(ProcessTree &tree, std::shared_ptr<EndpointSecurityAPI> esapi,
+                       const es_message_t *msg) {
```

Comment:
I think we need to consider taking taking a `const Message &` here, which already wraps the appropriate `EndpointSecurityAPI` instance (especially looking ahead at future PRs where this function will be called from...). The problem tho is that there are not appropriate functions exposed from `Message` that allow you to meaningfully act upon the contained `EndpointSecurityAPI`.

We should chat a bit about a good path forward here.

Reply:
after some discussion, I implemented the "basic" solution in [61701251e786dbb22800cecc64ef5769c5f93795](https://github.com/google/santa/pull/1237/commits/61701251e786dbb22800cecc64ef5769c5f93795) (exposing an ESAPI() method on `Message` and taking a `Message&` in `InformFromESEvent`).

I experimented making this interface nicer with some message subclasses that expose ESAPI wrapper methods as appropriate in https://github.com/kallsyms/santa/commit/2319da99c2a69ab54f21e3f4442e17b4199ea1fc which looks pretty nice to me, but is a separate concern from this PR.